pinot-plugins/pinot-input-format/pinot-orc/src/main/java/org/apache/pinot/plugin/inputformat/orc/ORCRecordReader.java

```diff
@@ -106,6 +110,19 @@ public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable Re
     _nextRowId = 0;
   }
 
+  private File unzipIfRequired(File dataFile) throws IOException {
+    if (dataFile.getName().endsWith(".gz")) {
```

Comment:
I think that the better approach is to add a helper function in `RecordReaderUtils` to identify whether the file is `gzipped` or not instead of depending on the extension. I have seen the cases where people deal with gzipped files (but not end with `.gz`).